libfreerdp/codec/CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+# codec
+
+set(CODEC_SRCS
+	bulk.c
+	bulk.h
+	dsp.c
+	color.c
+	color.h
+	audio.c
+	planar.c
+	bitmap.c
+	interleaved.c
+	progressive.c
+	rfx_bitstream.h
+	rfx_constants.h
+	rfx_decode.c
+	rfx_decode.h
+	rfx_differential.h
+	rfx_dwt.c
+	rfx_dwt.h
+	rfx_encode.c
+	rfx_encode.h
+	rfx_quantization.c
+	rfx_quantization.h
+	rfx_rlgr.c
+	rfx_rlgr.h
+	rfx_types.h
+	rfx.c
+	region.c
+	nsc.c
+	nsc_encode.c
+	nsc_encode.h
+	nsc_types.h
+	ncrush.c
+	xcrush.c
+	mppc.c
+	zgfx.c
+	clear.c
+	jpeg.c
+	h264.c
+	yuv.c)
+
+set(CODEC_SSE2_SRCS
+	sse/rfx_sse2.c
+	sse/rfx_sse2.h
+	sse/nsc_sse2.c
+	sse/nsc_sse2.h
+)
+
+set(CODEC_NEON_SRCS
+	neon/rfx_neon.c
+	neon/rfx_neon.h
+	neon/nsc_neon.c
+	neon/nsc_neon.h
+)
+
+# Append initializers
+set(CODEC_LIBS "")
+list(APPEND CODEC_SRCS ${CODEC_SSE2_SRCS})
+list(APPEND CODEC_SRCS ${CODEC_NEON_SRCS})
+
+if(WITH_SSE2)
+	if(CMAKE_COMPILER_IS_GNUCC OR ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
+		if (CODEC_SSE2_SRCS)
+			set_source_files_properties(${CODEC_SSE2_SRCS} PROPERTIES COMPILE_FLAGS "-msse2" )
+		endif()
+	endif()
+
+	if(MSVC)
+		if (CODEC_SSE2_SRCS)
+			set_source_files_properties(${CODEC_SSE2_SRCS} PROPERTIES COMPILE_FLAGS "/arch:SSE2" )
+		endif()
+	endif()
+endif()
+if(WITH_NEON)
+	check_symbol_exists("_M_AMD64"     ""  MSVC_ARM64)
```

Comment:
`_M_AMD64` is only defined for `ARM64EC` targets (ARM64 Emulation Compatible).
Maybe we should check for `_M_ARM64` here (and maybe also `_M_ARM64EC`)?

Reply:
will address this in an upcoming pr.